web-local/src/lib/components/workspace/explore/metrics-container/MeasuresContainer.svelte

```diff
@@ -0,0 +1,139 @@
+  const MEASURE_WIDTH = 175;
+  const MARGIN_TOP = 20;
+  const CHARACTER_LIMIT_FOR_WRAPPING = 26;
+
+  let metricsExplorer: MetricsExplorerEntity;
+  $: metricsExplorer = $metricsExplorerStore.entities[metricViewName];
+
+  $: instanceId = $runtimeStore.instanceId;
+
+  // query the `/meta` endpoint to get the measures and the default time grain
+  $: metaQuery = useMetaQuery(instanceId, metricViewName);
+
+  $: selectedMeasureNames = metricsExplorer?.selectedMeasureNames;
+
+  function getMeasureHeightsForColumn(measuresHeight, numColumns) {
+    const recalculatedHeights = [...measuresHeight];
+    for (let i = 0; i < measuresHeight.length; i = i + numColumns) {
+      const row = measuresHeight.slice(i, i + numColumns);
+      if (row.indexOf(MEASURE_HEIGHT_MULTILINE) != -1) {
+        for (let j = i; j < i + numColumns && j < measuresHeight.length; j++) {
+          recalculatedHeights[j] = MEASURE_HEIGHT_MULTILINE;
+        }
+      }
+    }
+    return recalculatedHeights;
+  }
+
+  let totalsQuery: UseQueryStoreResult<V1MetricsViewTotalsResponse, Error>;
+  $: numColumns = 1;
+  let measureGridHeights = [];
+
+  $: if (
+    metricsExplorer &&
+    metaQuery &&
+    $metaQuery.isSuccess &&
+    !$metaQuery.isRefetching
+  ) {
+    let totalsQueryParams = {
+      measureNames: selectedMeasureNames,
+      filter: metricsExplorer?.filters,
+    };
+
+    totalsQuery = useRuntimeServiceMetricsViewTotals(
+      instanceId,
+      metricViewName,
+      totalsQueryParams
+    );
+
+    const measures = $metaQuery.data?.measures;
+
+    let measuresHeight = measures.map((measure) => {
+      const label = measure?.label || measure?.expression;
+      if (label.length > CHARACTER_LIMIT_FOR_WRAPPING)
+        return MEASURE_HEIGHT_MULTILINE;
+      else return MEASURE_HEIGHT;
+    });
+
+    const totalMeasuresHeight = measuresHeight.reduce(
+      (s, v) => s + v + MARGIN_TOP,
+      0
+    );
+
+    if (metricsContainerHeight) {
+      const columns = totalMeasuresHeight / metricsContainerHeight;
+      if (columns <= 1) {
+        numColumns = 1;
+        measureGridHeights = [...measuresHeight];
+      } else if (columns > 2) {
+        numColumns = 3;
+        measureGridHeights = getMeasureHeightsForColumn(measuresHeight, 3);
+      } else {
+        numColumns = 2;
+        measureGridHeights = getMeasureHeightsForColumn(measuresHeight, 2);
+
+        // Check if two columns can individually accommodate all measures without scroll
+        const measuresHeightInSingleColumn = measureGridHeights
+          .filter((_, i) => i % 2 == 0)
+          .reduce((s, v) => s + v + MARGIN_TOP, 0);
+        const extraHeight =
+          metricsContainerHeight - measuresHeightInSingleColumn;
+        if (extraHeight < 0) {
+          numColumns = 3;
+          measureGridHeights = getMeasureHeightsForColumn(measuresHeight, 3);
+        }
+      }
+    }
+  }
+</script>
+
+<div class="grid grid-cols-{numColumns} gap-x-1">
```

Comment:
```suggestion
<div class="grid grid-cols-{numColumns} gap-x-4">
```

It looks a little squished when you have two measure labels next to each other that are very long